pkg/collect/configmap.go

```diff
@@ -0,0 +1,147 @@
+package collect
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type ConfigMapOutput struct {
+	Namespace       string `json:"namespace"`
+	Name            string `json:"name"`
+	Key             string `json:"key"`
+	ConfigMapExists bool   `json:"configMapExists"`
+	KeyExists       bool   `json:"keyExists"`
+	Value           string `json:"value,omitempty"`
+}
+
+func ConfigMap(c *Collector, configMapCollector *troubleshootv1beta2.ConfigMap) (map[string][]byte, error) {
+	client, err := kubernetes.NewForConfig(c.ClientConfig)
```

Comment:
we should REALLY refactor troubleshoot at some point to pass in a client instead of generating one each time

(this would also allow for better testing)

Reply:
yep

Reply:
It would also result in not having the `wall of shame` when using doctl:
```
W0708 13:07:14.096840    5597 exec.go:282] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network connections; 3211 clients constructed calling "doctl"
```